components/l10n/common/ofac_sanction_util.h

```diff
@@ -0,0 +1,32 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_L10N_COMMON_OFAC_SANCTION_UTIL_H_
+#define BRAVE_COMPONENTS_L10N_COMMON_OFAC_SANCTION_UTIL_H_
+
+#include <string>
+
+namespace brave_l10n {
+
+// Returns |true| if the given |country_code| is on the OFAC sanctioned list,
+// otherwise |false|. |country_code| supports ISO 3166-1 alpha-2 country codes,
+// ISO 3166-1 alpha-3 country codes, and UN M49 area codes. Set
+// |should_sanction_un_m49_area_codes| to |true| to sanction UN M49 area codes.
+//
+// NOTE: Sanctioning UN M49 area codes will also block all other countries that
+// are part of the same area code.
+//
+// See:
+//
+//   https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2
+//   https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3
+//   https://en.wikipedia.org/wiki/ISO_3166-1_numeric
+//   https://en.wikipedia.org/wiki/UN_M49
+bool IsCountryCodeOFACSanctioned(const std::string& country_code,
+                                 bool should_sanction_un_m49_area_codes);
```

Comment:
This parameter seems a bit confusing to me. IIUC, M49 area codes are different from the 3-digit ISO-3166 codes. The caller would have to know whether they are holding an M49 or ISO code, correct? If so, would it be clearer to have a separate function for M49? (e.g. `IsAreaCodeOFACSanctioned`)

Reply:
Agreed, thanks. I have separated both ISO and UN M49 checks.